### PR TITLE
Borg buffs - Allows jaws to incapacitate, gives secborg a taser, and more! (wall of text in desc)

### PR DESCRIPTION
**Pull request by deathride58**
## About The Pull Request
So, this PR hinges primarily on #9466 . If that PR is closed, then the WEAKEN_SECBORG config option will be made the default. Basically, this makes K9s require more skill to play, but also makes it more rewarding. This PR allows dogborg jaws to incapacitate (with a weaker variant for puppy jaws), gives the standard secborg a taser, allows borg flashes to knockdown, makes some adjustments to dogborg pounces, and reduces the base health of K9 borgs. The first three things listed can be toggled off in the config. This PR further solidifies the K9's role as a high-risk high-reward module, and generally brings the strength of both security modules up to compensate for their impending fate as red-alert-only modules. The two new config options added allows this behavior to be configured at runtime, while providing downstreams and other server operators an easy option to mostly retain the original behavior.

The dogborg jaw incapacitation allows K9s and other dogborgs to latch onto targets, which causes a hard stun on the target depending on their staminaloss and resting status, up to a maximum of 5.6 seconds for K9 jaws, and 2.8 seconds for puppy jaws. This also causes the dogborg to grab the target with a lower click delay than normal grabs. If performed against a standing target, this will cause a knockdown with a strength double that of the target's current staminaloss (meaning it only functions as a grab against targets with no staminaloss). If the stun is long enough for the target to be successfully cuffed before the stun is over, a far meatier sound will play when the grab goes through. This is behavior toggled by the WEAKEN_SECBORG config option. The optimal way to make use of this new feature is to either shoot your target with a disabler a few times or get them to exhaust themself by running, pounce on them once they're at about half stamina, then quickly follow the pounce up with a bite, and quickly switch to your cuffs to slap 'em on before the stun is over. This combo requires quite a bit of skill and a stable internet connection to pull off, but if pulled off right, will get the target begging for dogborgs to be nerfed before the round is over, as not even adrenals will save you from the chomp.

Speaking of K-9s, the pounce has also been tweaked. It can no longer hard stamcrit the target on its own, and can only bring the target up to a maximum of 120 staminaloss, and can deal a maximum of 80 stamloss. Additionally, the spoolup has been brought back to its original value of half a second, and telegraphing has been added in the form of a quiet servo revving sound (not bringing back the eye glint unless people really want it). The cooldown has also been brought up to 2.5 seconds (cooldown starts counting down the moment you click, it's 3 seconds in code) to encourage being more resourceful with your pounce.

K9s have also had their base health decreased to 80. This is to help further cement the intended high-risk high-reward gameplay style of the K9 by acting as a counterbalance to the new mechanics. Without this nerf, K9s would probably be a little too strong even for a red-alert exclusive borg type.

Secborgs getting a taser is self-explanatory. This allows secborgs to play a more aggressive role against station threats by giving them back the ability to knockdown from a distance. Whether or not a secborg will get a hybrid taser or just a disabler can be toggled via the WEAKEN_SECBORG config option.

The disabler cooler upgrade has also been buffed to apply to every energy-based firearm in a borg's arsenal instead of just the disabler. This is self-explanatory, and allows the upgrade to continue to apply to secborgs.

Borg flashes, and only borg flashes, have regained the ability to knockdown carbon mobs! This is primarily to give borgs a much better option to deal with troublemakers, as a way to compensate for the loss of secborgs outside of red alert. This can be disabled via the DISABLE_BORG_FLASH_KNOCKDOWN config option.

Again, this PR is intended to go with #9466 . If #9466 is closed, then this PR will also be closed, as this PR puts secborgs in way too strong of a position balance-wise for these changes to be reasonable without that PR. The reason why this is atomized is to allow these changes to be reviewed, testmerged, and analyzed independently of the security level restriction mechanic.

## Changelog
:cl: Bhijn
tweak: K9 pounces have received a minor rework. It now has an effective cooldown of 2.5 seconds, can now only deal up to 120 staminaloss, deals a maximum of 80 stamloss on hit, has a spoolup of half a second, and now has telegraphing in the form of a quiet noise.
balance: K9s now only have 80 health
balance: Secborgs (but not k9s) now have a hybrid taser. This can be toggled via server config.
tweak: The disabler cooler upgrade now applies to all energy-based firearms for borgs
add: Dogborg jaws are now capable of incapacitating targets if using help intent. This deals a hard stun depending on how much staminaloss the target has, and whether or not they're resting. This behavior can be toggled via server config.
balance: K9 jaws now have 15 force, up from their nerfed 10 force.
balance: Borg flashes regained their ability to cause knockdown. This can be toggled via server config.
server: The WEAKEN_SECBORG config option will disable the new dogborg jaws mechanic and make secborgs spawn with a standard disabler.
server: The DISABLE_BORG_FLASH_KNOCKDOWN will disable the ability for borg flashes to knockdown.
/:cl:

### CODE REVIEW

**Comment by GrayRachnid** [comment]
I hate this just because it gives borgs tasers again despite being a secborg player. Remove tasers, they're bad for combat.

File: modular_citadel/code/modules/mob/living/silicon/robot/dogborg_equipment.dm [hunk 1]
```diff
@@ -307,8 +346,10 @@ SLEEPER CODE IS IN game/objects/items/devices/dogborg_sleeper.dm !
 /mob/living/silicon/robot
 	var/leaping = 0
 	var/pounce_cooldown = 0
-	var/pounce_cooldown_time = 20 //Buffed to counter balance changes
-	var/pounce_spoolup = 1
+	var/pounce_cooldown_time = 30 //Time in deciseconds between pounces
+	var/pounce_spoolup = 5 //Time in deciseconds for the pounce to happen after clicking
+	var/pounce_stamloss_cap = 120 //How much staminaloss pounces alone are capable of bringing a spaceman to
+	var/pounce_stamloss = 80 //Base staminaloss value of the pounce
```
Comment (GrayRachnid) [inline modular_citadel/code/modules/mob/living/silicon/robot/dogborg_equipment.dm:352]: I think this should be brought down to 60.

**Comment by Strazyplus** [comment]
red-alert-only module.
Why?

**Comment by BlackMajor** [comment]
> red-alert-only module.
> Why?

https://github.com/Citadel-Station-13/Citadel-Station-13/pull/9466

**Comment by Strazyplus** [comment]
> > red-alert-only module.
> > Why?
> 
> #9466

Well I enjoyed being secborg, it was fun. but the hate is just getting stupid, although this change does not completely remove them it severely makes them disappear. as when things go red the roboticist is likely to focus on his time on mechs than that borg wanting a reset.

File: code/modules/assembly/flash.dm [hunk 2]
```diff
@@ -201,8 +201,10 @@
 /obj/item/assembly/flash/cyborg
 
 /obj/item/assembly/flash/cyborg/attack(mob/living/M, mob/user)
-	..()
+	. = ..()
 	new /obj/effect/temp_visual/borgflash(get_turf(src))
+	if(. && !CONFIG_GET(flag/disable_borg_flash_knockdown) && iscarbon(M) && !M.resting)
+		M.Knockdown(80)
```
Comment (Poojawa) [inline code/modules/assembly/flash.dm]: This should be atomized, not rolled into your red alert specific garbage.

File: modular_citadel/code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 3]
```diff
@@ -53,6 +53,7 @@
 	has_snowflake_deadsprite = TRUE
 	dogborg = TRUE
 	cyborg_pixel_offset = -16
+	borghealth = 80
```
Comment (Poojawa) [inline modular_citadel/code/modules/mob/living/silicon/robot/robot_modules.dm:56]: Why the K9 unit susceptible to being completely fucking disabled with one shotgun blast? or 357 shot or 3 shots from blasters?

The fuck kind of balance is that.

**Comment by Soranayo** [comment]
Honestly, i do like the idea of K9 and normal secborgs having a different ways to arrest people, the health might be a little low for a module that requires you to get up close and personal, in my opinion.

**Comment by xurxomario** [comment]
Why is the borg technically designed to protect crew (technically all are but you know what i mean) from literal red alert threats (read: the only kind of threats that will actively go and kill security, and are armed to do so) getting a nerf in health? Right after a nerf that dissallows them to come out before that threat gets announced? What i think will come out of this is one of two things, either people red alerting on a quicker draw finger to have more secborgs to compensate for how utterly frail they will be, or, the other option, which i find theres more chance of happening, is loud red alert antags just rushing and deswording/bombing/killing the roboticists as soon as they set foot on station, and i dunno if making robotics more of a target than it already was is really a wise idea, as getting murderboned 7 rounds out of 10 is sure to drive some of the competent roboticists away, especially with how slow robotics can be with no upgrades.

Also i was gonna to propose an alternate balance but i lost focus and forgot haha, yea, i was gonna say, maybe increase the life to 100 again, or even 125 or something, cause its a metal hunk, it should be hard to beat down, and at the same time apply a multiplier to burn damage (x2 or something), making EMPs a more reliable weapon aganist them.

**Review by silicons — APPROVED** [review]
